pkgs/development/libraries/openimagedenoise/default.nix

```diff
@@ -23,14 +35,39 @@ stdenv.mkDerivation rec {
 
   patches = lib.optional cudaSupport ./cuda.patch;
 
-  nativeBuildInputs = [
-    cmake
-    python3
-    ispc
-  ] ++ lib.optional cudaSupport cudaPackages.cuda_nvcc;
+  # metal support depends on the "metal" framework from xcode, which isn't packed in the apple sdks in nixpkgs
+  postPatch =
+    ''
+      substituteInPlace devices/metal/CMakeLists.txt \
+        --replace-fail "AppleClang" "Clang"
+    ''
+    + lib.optionalString (appleMetalSupport) ''
+      substituteInPlace cmake/oidn_metal.cmake \
+        --replace-fail "xcrun -sdk macosx metal" "${darwin.xcode_14}/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/metal"
+    '';
```

Comment:
It just took me quite some time to figure this out..It would be a shame to just remove it^^
But sure, it isn't testable as is now and I'm fine either way